api/internal/snapshot/snapshot.go

```diff
@@ -49,6 +49,7 @@ func (service *Service) stop() {
 	}
 
 	close(service.refreshSignal)
+	service.refreshSignal = nil
```

Comment:
@chiptus could you share some information about how this solve the issue?

Reply:
because `.Start()` won't start unless `refreshSignal` is nil.
(that means that on update, the process stops - `close(refreshSignal)` but don't restart) 
also `stop` fails when `refreshSignal` is not nil but closed.

so we close the last signal and delete it. this makes both start and stop work.

Reply:
```suggestion
        // .Start() won't start unless refreshSignal is nil.
        // (that means that on update, the process stops - close(refreshSignal) but don't restart)
        // also stop fails when refreshSignal is not nil but closed.
        // so we close the last signal and delete it. this makes both start and stop work
	service.refreshSignal = nil
```

Reply:
added a shorter comment